fusestore/store/src/engine/mem_engine.rs

```diff
@@ -38,8 +38,12 @@ impl MemEngine {
     ) -> anyhow::Result<i64> {
         // TODO: support plan.engine plan.options
         let curr = self.dbs.get(&cmd.db_name);
-        if curr.is_some() && if_not_exists {
-            return Err(anyhow::anyhow!("database exists"));
+        if let Some(curr) = curr {
+            return if if_not_exists {
+                Ok(curr.db_id)
+            } else {
+                Err(anyhow::anyhow!("{} database exists", cmd.db_name))
+            };
```

Comment:
If the DB exists, and `if_not_exists==true`, shouldn't it return an error? 
Without an error, the caller does not know whether a new DB is created or not.

And when `if_not_exists==false`, I think it means no matter the DB exists or not, it should always create a new DB but not returning an error. 🤔 ?

Reply:
See:
https://dev.mysql.com/doc/refman/8.0/en/create-database.html
An error occurs if the database exists and you did not specify IF NOT EXISTS.

If we do `create db/table` and won't get an error, we can use it.